### ISSUE TITLE
Error: Failed to execute 'put' on 'IDBObjectStore': The object store uses in-line keys and the key parameter was provided. when I try to use updateByKey 

### ISSUE THREAD

**Issue by muzafarali** [assuncaocharles/ngx-indexed-db#273]
Hi,
I am getting error when I try to use updateByKey

> let data = {
>   tab_3: formData['tab_3']
> }
> const store$ = this.dbService.updateByKey('transformer_safety_history', data, 4);

I have also try different combination like 

> let data = {
>   tab_3: formData['tab_3'],
>   id: 4
> }
> const store$ = this.dbService.updateByKey('transformer_safety_history', data);

and when I try to use 

`const store$ = this.dbService.update('transformer_safety_history', data);`

its removed all old data and add new data that are post

here is error log

> 2021-09-03T07:07:37.659Z ERROR [main.js:20081] Error: Failed to execute 'put' on 'IDBObjectStore': The object store uses in-line keys and the key parameter was provided.
>     at http://localhost:4200/vendor.js:226422:29
>     at ZoneDelegate.invoke (http://localhost:4200/polyfills.js:683:26)
>     at Object.onInvoke (http://localhost:4200/vendor.js:57382:33)
>     at ZoneDelegate.invoke (http://localhost:4200/polyfills.js:682:52)
>     at Zone.run (http://localhost:4200/polyfills.js:445:43)
>     at http://localhost:4200/polyfills.js:1587:36
>     at ZoneDelegate.invokeTask (http://localhost:4200/polyfills.js:717:31)
>     at Object.onInvokeTask (http://localhost:4200/vendor.js:57369:33)
>     at ZoneDelegate.invokeTask (http://localhost:4200/polyfills.js:716:60)
>     at Zone.runTask (http://localhost:4200/polyfills.js:489:47)

Here is version details:

> 
> "@angular/common": "11.2.1",
>     "@angular/compiler": "11.2.1",
>     "@angular/core": "^11.2.1",
> "ngx-indexed-db": "^9.3.0",

**Comment by CArnaboldi** [comment]
I can confirm I'm having exactly the same issue.

Given the following store config:

```
      {
        store: storicoDifettiStore,
        storeConfig: { keyPath: 'GUID', autoIncrement: false },
        storeSchema: [
        ]
      },
```

When I issue the command ("toSaveOrUpdate" object already contains the GUID...it's an update after all..):

`this.localDBServ.updateByKey<StoricoDifettoDB>(storicoDifettiStore, toSaveOrUpdate, toSaveOrUpdate.GUID);`

I get the same error:

**core.js:6141 ERROR DOMException: Failed to execute 'put' on 'IDBObjectStore': The object store uses in-line keys and the key parameter was provided.**

Edit: at the moment I'm using the following workaround (first delete, then make a new insert..):

```
      return this.localDBServ.deleteByKey(storicoDifettiStore, toSaveOrUpdate.GUID)
        .pipe(
          switchMap(_ => this._addEntity<StoricoDifettoDB>(storicoDifettiStore, toSaveOrUpdate))
        );
```

where "_addEntity" method does the following (required because "addItem" method is also bugged, see issue #263 ):

```
  private _addEntity<T>(storeName: string, entity: T): Observable<T>{
      return this.localDBServ.bulkAdd<T[]>(storeName, [entity])
          .pipe(
              switchMap(result => this.localDBServ.getByKey<T>(storeName, result[0]))
          );
  }
```

**Comment by muzafarali** [comment]
Here is my code
```
{
      store: 'transformer_safety_history',
      storeConfig: { keyPath: 'id', autoIncrement: false },
      storeSchema: [
        { name: 'location_id', keypath: 'location_id', options: { unique: false } },
        { name: 'transformer_safety_id', keypath: 'transformer_safety_id', options: { unique: false } },
        { name: 'user_id', keypath: 'user_id', options: { unique: false } },
        { name: 'tab_1', keypath: 'tab_1', options: { unique: false } },
        { name: 'tab_2', keypath: 'tab_2', options: { unique: false } },
        { name: 'tab_3', keypath: 'tab_3', options: { unique: false } },
        { name: 'tab_4', keypath: 'tab_4', options: { unique: false } },
        { name: 'tab_5', keypath: 'tab_5', options: { unique: false } },
        { name: 'tab_6', keypath: 'tab_6', options: { unique: false } },
        { name: 'process', keypath: 'process', options: { unique: false } },
        { name: 'signature', keypath: 'signature', options: { unique: false } },
        { name: 'is_complete', keypath: 'is_complete', options: { unique: false } },
        { name: 'isOnServer', keypath: 'isOnServer', options: { unique: false } }
      ]
    },
```

here is im using to update
```
const store$ = this.dbService.updateByKey<TransformerSafetyHistoryDB>('transformer_safety_history', formData, formData.id);
```

formData is object of TransformerSafetyHistoryDB

and getting same error

```
ngx-logger.js:709 2021-09-03T10:16:21.245Z ERROR [main.js:20199] Error: Failed to execute 'put' on 'IDBObjectStore': The object store uses in-line keys and the key parameter was provided.
    at http://localhost:4200/vendor.js:226422:29
    at ZoneDelegate.invoke (http://localhost:4200/polyfills.js:683:26)
    at Object.onInvoke (http://localhost:4200/vendor.js:57382:33)
    at ZoneDelegate.invoke (http://localhost:4200/polyfills.js:682:52)
    at Zone.run (http://localhost:4200/polyfills.js:445:43)
    at http://localhost:4200/polyfills.js:1587:36
    at ZoneDelegate.invokeTask (http://localhost:4200/polyfills.js:717:31)
    at Object.onInvokeTask (http://localhost:4200/vendor.js:57369:33)
    at ZoneDelegate.invokeTask (http://localhost:4200/polyfills.js:716:60)
    at Zone.runTask (http://localhost:4200/polyfills.js:489:47)
_logModern @ ngx-logger.js:709
(anonymous) @ ngx-logger.js:807
__tryOrUnsub @ Subscriber.js:183
next @ Subscriber.js:122
_next @ Subscriber.js:72
next @ Subscriber.js:49
notifyNext @ switchMap.js:70
_next @ InnerSubscriber.js:11
next @ Subscriber.js:49
(anonymous) @ subscribeToArray.js:3
_trySubscribe @ Observable.js:42
subscribe @ Observable.js:28
subscribeToResult @ subscribeToResult.js:9
_innerSub @ switchMap.js:46
_next @ switchMap.js:36
next @ Subscriber.js:49
(anonymous) @ subscribeToArray.js:3
_trySubscribe @ Observable.js:42
subscribe @ Observable.js:28
call @ switchMap.js:17
subscribe @ Observable.js:23
_log @ ngx-logger.js:757
error @ ngx-logger.js:610
(anonymous) @ clickrun.component.ts:909
__tryOrUnsub @ Subscriber.js:183
error @ Subscriber.js:135
_error @ Subscriber.js:75
error @ Subscriber.js:55
_error @ tap.js:56
error @ Subscriber.js:55
notifyError @ OuterSubscriber.js:7
_error @ InnerSubscriber.js:14
error @ Subscriber.js:55
(anonymous) @ ngx-indexed-db.js:363
invoke @ zone.js:372
onInvoke @ core.js:28535
invoke @ zone.js:371
run @ zone.js:134
(anonymous) @ zone.js:1276
invokeTask @ zone.js:406
onInvokeTask @ core.js:28522
invokeTask @ zone.js:405
runTask @ zone.js:178
drainMicroTaskQueue @ zone.js:582
invokeTask @ zone.js:491
invokeTask @ zone.js:1600
globalZoneAwareCallback @ zone.js:1626
IndexedDB (async)
(anonymous) @ ngx-indexed-db.js:12
ZoneAwarePromise @ zone.js:1387
openDatabase @ ngx-indexed-db.js:8
(anonymous) @ ngx-indexed-db.js:350
_trySubscribe @ Observable.js:42
subscribe @ Observable.js:28
subscribeToResult @ subscribeToResult.js:9
_innerSub @ switchMap.js:46
_next @ switchMap.js:36
next @ Subscriber.js:49
_subscribe @ BehaviorSubject.js:14
_trySubscribe @ Observable.js:42
_trySubscribe @ Subject.js:81
subscribe @ Observable.js:28
call @ switchMap.js:17
subscribe @ Observable.js:23
call @ tap.js:16
subscribe @ Observable.js:23
processTab5 @ clickrun.component.ts:887
updateForkliftTransport @ clickrun.component.ts:462
(anonymous) @ clickrun.component.html:544
handleEvent @ core.js:31564
handleEvent @ core.js:32116
dispatchEvent @ core.js:23506
(anonymous) @ core.js:30769
(anonymous) @ platform-browser.js:582
invokeTask @ zone.js:406
onInvokeTask @ core.js:28522
invokeTask @ zone.js:405
runTask @ zone.js:178
invokeTask @ zone.js:487
invokeTask @ zone.js:1600
globalZoneAwareCallback @ zone.js:1626
Show 50 more frames

```

**Comment by vahid18u** [comment]
I have same issue when I provide a key in updateByKey method. the README.md example is not passing any key while the key is not optional.

**Comment by DadvDadv** [comment]
The same here...

There is a documentation error on that : [doc](https://github.com/assuncaocharles/ngx-indexed-db#updatebykeystorename-string-value-t-key-idbvalidkey-observable) => bad sample

**Comment by aparzi** [comment]
Hi everyone @DadvDadv @vahid18u @Y3kRulez @muzafarali, 
this issue is solved with the merge #309.

**Comment by DadvDadv** [comment]
thx i'll try it

**Comment by assuncaocharles** [comment]
Fixed on #309
